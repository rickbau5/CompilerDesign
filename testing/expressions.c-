int main() {
    // Basic
    assign = ment;
    arr[1] = ay;
    ohne[2];
    structure.member = something;
    struct.x;
    struct.y.z = 'y';

    // Relative
    a == b;
    b != c;
    d <= e;
    f >= g;
    h < i;
    j > k;
}
