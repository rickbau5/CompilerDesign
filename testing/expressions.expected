Func main returns type int [line: 1]
!   Child: 1  Compound [line: 1]
!   !   Child: 1  Assign: = [line: 3]
!   !   !   Child: 0  Id: assign [line: 3]
!   !   !   Child: 1  Id: ment [line: 3]
!   !   Sibling: 0  Assign: = [line: 4]
!   !   !   Child: 0  Op: [ [line: 4]
!   !   !   !   Child: 0  Id: arr [line: 4]
!   !   !   !   Child: 1  Const: 1 [line: 4]
!   !   !   Child: 1  Id: ay [line: 4]
!   !   Sibling: 1  Op: [ [line: 5]
!   !   !   Child: 0  Id: ohne [line: 5]
!   !   !   Child: 1  Const: 2 [line: 5]
!   !   Sibling: 2  Assign: = [line: 6]
!   !   !   Child: 0  Op: . [line: 6]
!   !   !   !   Child: 0  Id: structure [line: 6]
!   !   !   !   Child: 1  Id: member [line: 6]
!   !   !   Child: 1  Id: something [line: 6]
!   !   Sibling: 3  Op: . [line: 7]
!   !   !   Child: 0  Id: struct [line: 7]
!   !   !   Child: 1  Id: x [line: 7]
!   !   Sibling: 4  Assign: = [line: 8]
!   !   !   Child: 0  Op: . [line: 8]
!   !   !   !   Child: 0  Op: . [line: 8]
!   !   !   !   !   Child: 0  Id: struct [line: 8]
!   !   !   !   !   Child: 1  Id: y [line: 8]
!   !   !   !   Child: 1  Id: z [line: 8]
!   !   !   Child: 1  Const: 'y' [line: 8]
!   !   Sibling: 5  Assign: = [line: 11]
!   !   !   Child: 0  Id: four [line: 11]
!   !   !   Child: 1  Op: * [line: 11]
!   !   !   !   Child: 0  Const: 4 [line: 11]
!   !   !   !   Child: 1  Const: 1 [line: 11]
!   !   Sibling: 6  Assign: = [line: 12]
!   !   !   Child: 0  Id: five [line: 12]
!   !   !   Child: 1  Op: / [line: 12]
!   !   !   !   Child: 0  Const: 15 [line: 12]
!   !   !   !   Child: 1  Const: 3 [line: 12]
!   !   Sibling: 7  Assign: = [line: 13]
!   !   !   Child: 0  Id: one [line: 13]
!   !   !   Child: 1  Op: % [line: 13]
!   !   !   !   Child: 0  Const: 3 [line: 13]
!   !   !   !   Child: 1  Const: 2 [line: 13]
!   !   Sibling: 8  Assign: = [line: 16]
!   !   !   Child: 0  Id: two [line: 16]
!   !   !   Child: 1  Op: - [line: 16]
!   !   !   !   Child: 0  Const: 5 [line: 16]
!   !   !   !   Child: 1  Const: 3 [line: 16]
!   !   Sibling: 9  Assign: = [line: 17]
!   !   !   Child: 0  Id: four [line: 17]
!   !   !   Child: 1  Op: + [line: 17]
!   !   !   !   Child: 0  Id: two [line: 17]
!   !   !   !   Child: 1  Const: 2 [line: 17]
!   !   Sibling: 10  Assign: = [line: 20]
!   !   !   Child: 0  Id: neg [line: 20]
!   !   !   Child: 1  Op: - [line: 20]
!   !   !   !   Child: 0  Const: 1 [line: 20]
!   !   Sibling: 11  Assign: = [line: 21]
!   !   !   Child: 0  Id: mul [line: 21]
!   !   !   Child: 1  Op: * [line: 21]
!   !   !   !   Child: 0  Const: 6 [line: 21]
!   !   Sibling: 12  Assign: = [line: 22]
!   !   !   Child: 0  Id: question [line: 22]
!   !   !   Child: 1  Op: ? [line: 22]
!   !   !   !   Child: 0  Const: 4 [line: 22]
!   !   Sibling: 13  Assign: = [line: 23]
!   !   !   Child: 0  Id: multi [line: 23]
!   !   !   Child: 1  Op: - [line: 23]
!   !   !   !   Child: 0  Op: * [line: 23]
!   !   !   !   !   Child: 0  Op: ? [line: 23]
!   !   !   !   !   !   Child: 0  Id: yes [line: 23]
!   !   Sibling: 14  Assign: = [line: 26]
!   !   !   Child: 0  Id: ord [line: 26]
!   !   !   Child: 1  Op: or [line: 26]
!   !   !   !   Child: 0  Const: true [line: 26]
!   !   !   !   Child: 1  Const: false [line: 26]
!   !   Sibling: 15  Assign: = [line: 27]
!   !   !   Child: 0  Id: anded [line: 27]
!   !   !   Child: 1  Op: and [line: 27]
!   !   !   !   Child: 0  Const: false [line: 27]
!   !   !   !   Child: 1  Op: or [line: 27]
!   !   !   !   !   Child: 0  Const: true [line: 27]
!   !   !   !   !   Child: 1  Const: false [line: 27]
!   !   Sibling: 16  Assign: = [line: 28]
!   !   !   Child: 0  Id: notted [line: 28]
!   !   !   Child: 1  Op: not [line: 28]
!   !   !   !   Child: 0  Const: true [line: 28]
!   !   Sibling: 17  Op: == [line: 31]
!   !   !   Child: 0  Id: a [line: 31]
!   !   !   Child: 1  Id: b [line: 31]
!   !   Sibling: 18  Op: != [line: 32]
!   !   !   Child: 0  Id: b [line: 32]
!   !   !   Child: 1  Id: c [line: 32]
!   !   Sibling: 19  Op: <= [line: 33]
!   !   !   Child: 0  Id: d [line: 33]
!   !   !   Child: 1  Id: e [line: 33]
!   !   Sibling: 20  Op: >= [line: 34]
!   !   !   Child: 0  Id: f [line: 34]
!   !   !   Child: 1  Id: g [line: 34]
!   !   Sibling: 21  Op: < [line: 35]
!   !   !   Child: 0  Id: h [line: 35]
!   !   !   Child: 1  Id: i [line: 35]
!   !   Sibling: 22  Op: > [line: 36]
!   !   !   Child: 0  Id: j [line: 36]
!   !   !   Child: 1  Id: k [line: 36]
